src/v/ssx/sleep_abortable.h

```diff
@@ -44,6 +44,11 @@ sleep_abortable(typename Clock::duration dur, AbortSource&... src) {
         }
         seastar::weak_ptr<as_state> state;
     };
+
+    if ((src.abort_requested() || ...)) {
+        return seastar::make_exception_future<>(seastar::sleep_aborted());
```

Comment:
ahh nice. maybe the assumption by the author was that subscribing to an aborted source would inline invoke the callback, in which case i think this would have worked correctly since presumably sleep_abortable would check the abort source before sleeping.